src/java/org/apache/cassandra/service/accord/txn/TxnCondition.java

```diff
@@ -289,54 +291,99 @@ public boolean applies(TxnData data)
             FilteredPartition partition = reference.getPartition(data);
             boolean exists = partition != null && !partition.isEmpty();
 
-            Row row = null;
-            if (exists)
+            if (reference.column() != null
+                && !reference.column().isPartitionKey())
```

Comment:
Funny that the static analysis tools complain about this leaving NPE possible, but then I change it to...

```
ColumnMetadata column = reference.column();
if (column != null && !column.isPartitionKey())
```
...and it's apparently totally fine.

Reply:
This was a pet pev of mine in kotlin (the same people who made those tool you reference)... between the first and second call to `column` it might become null... by storing to a variable, you remove the possibility.

This is an issue with the static analysis, it doesn't really understand the callers and really doesn't grasp that this is basically immutable.  

I am 100% fine changing though, less warnings the better

Reply:
pushed the fix.  The code is more readable with the fix than without it, so its still a pro